src/interfaces/IOracles.sol

```diff
@@ -1,41 +1,142 @@
+        address[] calldata dataProviders
+    ) external;
+
+    /**
+     * @notice Removes a rate feed.
+     * @param rateFeedId The rate feed's ID.
+     * @dev Only callable by the owner.
+     */
+    function removeRateFeed(address rateFeedId) external;
+
+    /**
+     * @notice Adds a new trusted data provider, i.e. an oracle node operator
+     * @param rateFeedId The rate feed for which the new data provider is allowed to report.
+     * @param provider   The new data provider's address.
+     * @dev Only callable by the owner.
+     */
+    function addDataProvider(address rateFeedId, address provider) external;
+
+    /**
+     * @notice Removes a data provider from being allowed to report for a rate feed.
+     * @param rateFeedId The rate feed from which the data provider should be removed
+     * @param provider   The data provider's address.
+     * @dev Only callable by the owner.
+     */
+    function removeDataProvider(address rateFeedId, address provider) external;
+
+    /**
+     * @notice Sets validity flag `isFresh` to `0` for a rate feed if the most recent report has become outdated.
      * @param rateFeedId The rate feed to mark stale.
      */
     function markStale(address rateFeedId) external;
 
     /**
-     * @notice Sets the window size over which a rate feed's medians will be
-     * averaged.
+     * @notice Sets the price window size over which a rate feed's median prices will be averaged.
      * @param rateFeedId The rate feed being configured.
-     * @param windowSize The number of most recent medians to average over for
-     * the final reported median.
+     * @param priceWindowSize The number of most recent median prices to average over for the final reported price.
+     * @dev For example, if `priceWindowSize` was 3 and the latest 5 median prices were [1, 3, 2, 3, 4], then we would
+     *      average over the last three values [2, 3, 4] to get the final reported average price of (2 + 3 + 4) / 3 = 3
      * @dev Only callable by the owner.
      */
-    function setWindowSize(address rateFeedId, uint8 windowSize) external;
+    function setPriceWindowSize(
+        address rateFeedId,
+        uint8 priceWindowSize
+    ) external;
 
     /**
-     * @notice Sets the allowed deviation for a rate feed.
+     * @notice Sets the allowed price deviation between the lowest and highest price values in a report for a rate feed.
      * @param rateFeedId The rate feed being configured.
-     * @param allowedDeviation The maximal multiplicative deviation allowed
-     * between two values in a report batch, expressed as the numerator of a
-     * fraction over uint16.max.
+     * @param allowedDeviation The difference between the lowest and highest value in a price report.
+     *        Expressed as the numerator of a fraction over uint16.max (65535). I.e., if allowedDeviation was 10_000,
+     *        then the difference between price values can't be greater than 1_000 / 65_535 = 0.015259... ≈ 1.526%
```

Comment:
I think the intended mechanism is a bit different: we set a high `allowedDeviation` to restrict the allowed ratio. The value restricts how small the ratio between the lowest and highest price can be. E.g. if it's set to `65_530`, then the lowest value divided by the highest value has to be very close to 1. At least that's how I was thinking about this parameter and its usage.

Reply:
Aha, then I have misunderstood it, apparently!

It seems to me they'd be functionally equivalent, though, just different way of phrasing it, no?

My understanding of `allowedDeviation` was: _"the highest and lowest value in a price report can not differ by more than `allowedDeviation/65_535` in %"_

Your intent AFAIU was this:

```
@param allowedRatio The allowed difference between the lowest and highest value in a price report.
     *        Expressed as the numerator of a fraction over uint16.max (65_535). I.e., if allowedRatio was 60_000,
     *        then the ratio of the lowest value divided by the highest value in a price report can't be greater than
     *        60_000 / 65_535 = 0.991162137606343
@dev The higher we set `allowedRatio`, the lower the deviation tolerance between the lowest and highest price in a price report is
```

Thinking about this a bit more, to me personally `allowedDeviation` seems more obvious and intuitively auto-completes in my brain to "allowed deviation between prices in a report" whereas `allowedRatio` does not immediately click.

I think we could reconcile this by inverting the concept to mean `allowedDeviation = allowedDeviationNumerator/65_535`

Because intuitively the name suggests that the higher we set it => the higher our deviation tolerance is, and vice versa.

**Summing up, wdyt about this:**

```
@param allowedDeviationNumerator The allowed difference between the lowest and highest value in a price report.
     *        Expressed as the numerator of a fraction over uint16.max (65_535). I.e., if allowedRatio was 1_000,
     *        then the difference between price values can't be greater than 1_000 / 65_535 = 0.015259... ≈ 1.526%
@dev The higher this value is, the higher the deviation tolerance between the lowest and highest price in a price report is
```

Reply:
Sure, that could work as well. Just thinking about how the math would work, but I think it's still easy to do: multiply the lower number by `uint16.max`, divide by the higher number, compare to the parameter.

Reply:
So to formally define this: the deviation parameter `d` is the numerator of a fraction over `uint16.max` such that the maximal value of a batch (call it `M`) can't be larger than the minimal value (`m`) plus that fraction of `m`. Symbolically: `m * (1 + d/uint16.max) >= M`, or `m + m * (d/uint16.max) >= M`. So the fraction wouldn't be a percentage per se, so we don't need to include an additional division by 100 (as in the explanation in your comment).

Reply:
An alternative would be to use this as a percentage difference parameter, i.e. check for `(M - m) / ((M + m)/2) <= d/uint16.max`.

Reply:
I think in practice, when configuring new rate feeds, we will likely think & communicate in percent wrt max deviation we want to allow.

I'm wondering if we need the full uint16.max precision here or if giving up some precision for the sake of a simpler interface would make sense here.

I.e. instead of `d/uint16.max` we could make it `d/10_000` which would allow for a more intuitive handling.

I'm imagining the input from nadiem/roman to be "max deviation should be 0.5%" so the `allowedDeviation` param could just be `50` instead of `327` (327/65535 = 0.5%)